### PR TITLE
Fix issue where a new line was always written for the live log finish section

### PR DESCRIPTION
**Pull request by s0undt3ch**
This is a fix for the feature PR https://github.com/pytest-dev/pytest/pull/3189 where we'd always get 2 new lines between each log message on the finish section.

- [x] Target the `features` branch for new features and removals/deprecations.
- [x] Include new tests or update existing tests when applicable.

### CODE REVIEW

File: _pytest/logging.py [hunk 1]
```diff
@@ -489,14 +490,20 @@ def set_when(self, when):
...
     def emit(self, record):
         if self.capture_manager is not None:
             self.capture_manager.suspend_global_capture()
         try:
-            if not self._first_record_emitted or self._when in ('teardown', 'finish'):
+            if not self._first_record_emitted:
                 self.stream.write('\n')
                 self._first_record_emitted = True
+            elif self._when in ('teardown', 'finish'):
+                if self._test_outcome_written is False:
```
Comment (nicoddemus) [inline _pytest/logging.py]: It is considered more Pythonic to write `if not self._test_outcome_written` instead.

File: testing/logging/test_reporting.py [hunk 2]
```diff
@@ -293,6 +294,62 @@ def test_log_2(fix):
     ])
 
 
+def test_sections_single_new_line_after_test_outcome(testdir, request):
+    """Check that with live logging enable we are printing the correct headers during
```
Comment (nicoddemus) [inline testing/logging/test_reporting.py]: It seems this docstring is not related to the test

File: _pytest/logging.py (hunk 1 above)
Comment (s0undt3ch) [reply _pytest/logging.py]: But less explicit. Personal habits...

 I'll change.

File: testing/logging/test_reporting.py (hunk 2 above)
Comment (s0undt3ch) [reply testing/logging/test_reporting.py]: yeah, copy/paste

**Comment by s0undt3ch** [comment]
Updated.

**Comment by nicoddemus** [comment]
I tested out the branch and it works as advertised. 👍 

Thanks a lot @s0undt3ch!
